contrib/endpoints/src/api_manager/check_security_rules.cc

```diff
@@ -14,92 +14,350 @@
 //
 ////////////////////////////////////////////////////////////////////////////////
 #include "contrib/endpoints/src/api_manager/check_security_rules.h"
+#include <iostream>
+#include <sstream>
+#include "contrib/endpoints/src/api_manager/auth/lib/json_util.h"
+#include "contrib/endpoints/src/api_manager/proto/security_rules.pb.h"
+#include "contrib/endpoints/src/api_manager/utils/marshalling.h"
 
-#include <string>
-
-#include "contrib/endpoints/include/api_manager/api_manager.h"
-#include "contrib/endpoints/include/api_manager/request.h"
-
+using ::google::api_manager::auth::GetProperty;
+using ::google::api_manager::auth::GetStringValue;
 using ::google::api_manager::utils::Status;
+using ::google::protobuf::Map;
+using ::google::protobuf::util::error::Code;
 
 namespace google {
 namespace api_manager {
-
 namespace {
 
 const char kFirebaseServerStaging[] =
     "https://staging-firebaserules.sandbox.googleapis.com/";
 
+const char kFirebaseService[] =
+    "/google.firebase.rules.v1.FirebaseRulesService";
+
+const char kFailedFirebaseReleaseFetch[] = "Failed to fetch Firebase Release";
+const char kFailedFirebaseTest[] = "Failed to execute Firebase Test";
+const char kInvalidResponse[] = "Invalid JSON response from Firebase Service";
+
 // An AuthzChecker object is created for every incoming request. It does
 // authorizaiton by calling Firebase Rules service.
 class AuthzChecker : public std::enable_shared_from_this<AuthzChecker> {
```

Comment:
This was named by Limin and so was the file name provided by her. Let me see what she thinks too :-)